core/src/test/java/org/apache/iceberg/TestSchemaUpdate.java

```diff
@@ -364,7 +365,11 @@ public void testUpdateFailure() {
             Types.FixedType.ofLength(4),
             Types.DecimalType.of(9, 2),
             Types.DecimalType.of(9, 3),
-            Types.DecimalType.of(18, 2));
+            Types.DecimalType.of(18, 2),
+            Types.GeometryType.get(),
+            Types.GeometryType.of("srid:3857"),
+            Types.GeographyType.get(),
+            Types.GeographyType.of("srid:4269", EdgeInterpolationAlgorithm.KARNEY));
```

Comment:
Good catch with this test. @aihuaxu we will need to update this for variant and other types as well.